bib/internal/osbuildprogress/progress.go

```diff
@@ -0,0 +1,203 @@
+package osbuildprogress
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cheggaaa/pb/v3"
+)
+
+type OsbuildJsonProgress struct {
+	ID      string `json:"id"`
+	Context struct {
+		Origin   string `json:"origin"`
+		Pipeline struct {
+			Name  string `json:"name"`
+			Stage struct {
+				Name string `json:"name"`
+				ID   string `json:"id"`
+			} `json:"stage"`
+			ID string `json:"id"`
+		} `json:"pipeline"`
+	} `json:"context"`
+	Progress struct {
```

Comment:
We need to add "build_result" here if/once https://github.com/osbuild/osbuild/pull/1831 is merged